src/Creatable.js

```diff
@@ -37,8 +37,16 @@ export type Props = SelectProps & CreatableProps;
 
 const compareOption = (inputValue, option) => {
   const candidate = inputValue.toLowerCase();
+  
+  if (typeof option.value === 'string') {
+    return (
+      option.value.toLowerCase() === candidate ||
+      option.label.toLowerCase() === candidate
+    );
+  }
+  
   return (
-    option.value.toLowerCase() === candidate ||
+    option.value === candidate ||
```

Comment:
Perhaps this should be `==` rather than `===` as I assume `inputValue` will always be a string, even if its numeric